lib/utils/app_router.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:go_router/go_router.dart';
+import 'package:kayla_flutter_ic/api/storage/secure_storage.dart';
+import 'package:kayla_flutter_ic/utils/route_paths.dart';
+import 'package:kayla_flutter_ic/views/forget_password/forget_password_view.dart';
+import 'package:kayla_flutter_ic/views/home/home_view.dart';
+import 'package:kayla_flutter_ic/views/login/login_view.dart';
+import 'package:kayla_flutter_ic/views/survey_detail/survey_detail_view.dart';
+
+class AppRouter {
+  final SecureStorage _secureStorage;
+
+  AppRouter(this._secureStorage);
+
+  GoRouter get router => GoRouter(
+        initialLocation: RoutePath.home.path,
+        routes: <GoRoute>[
+          GoRoute(
+            path: RoutePath.home.screen,
+            name: RoutePath.home.name,
+            builder: (context, state) => const HomeView(),
+            routes: [
+              GoRoute(
+                path: RoutePath.surveyDetail.screenWithPathParams,
+                name: RoutePath.surveyDetail.name,
+                builder: (context, state) => SurveyDetailView(
+                  surveyId:
+                      state.params[RoutePath.surveyDetail.pathParam] ?? '',
+                ),
+              ),
+              GoRoute(
+                path: RoutePath.forgetPassword.screen,
+                name: RoutePath.forgetPassword.name,
+                builder: (context, state) => const ForgetPasswordView(),
+              ),
```

Comment:
I think Forget password screen should be in the route of login screen instead 🤔

Reply:
From my implementation, if the ForgetPasswordView inside LoginView's routes and then we push the ForgetPasswordView, the GoRoute throw error 😂 . The error show that it's wrong location.

Reply:
Let me check a bit 🙆‍♂️

Reply:
@nmint8m I see the problem. To be correct, the ForgetPassword screen should be a nested route of Login, so we will set the tree structure:
```
GoRoute(
            path: RoutePath.login.screen,
            name: RoutePath.login.name,
            builder: (context, state) => const LoginView(),
            routes: [
              GoRoute(
                path: RoutePath.forgetPassword.screen,
                name: RoutePath.forgetPassword.name,
                builder: (context, state) => const ForgetPasswordView(),
              ),
            ],
          ),
```
In the `redirect` function the condition is not correct all the time IMO, because `sublocation` will return the full path of the route, so in this case, forgetPassword path will be: `/login/forget-password`.
I think we should update the redirect function a bit:
```
if (!isLogin &&
              !subLocation.endsWith(RoutePath.forgetPassword.path) &&
              !subLocation.endsWith(RoutePath.login.path)) {
            return RoutePath.login.path;
          }
```

Reply:
@manh-t 

My purpose is to keep the forget password screen has the location `/forget-password`, instead of `/login/forget-password`. So I don't want to use the function `push` the forget password screen into the login screen. Alternatively, I use `go` function.

Base on your suggestion, I will move the forget password screen to the same level as the home screen and the login screen. Sounds good to you?

The idea is that:

- Home ↔ Login: Navigation by using `go` function, the location will be `/` ↔ `/login`
- Login ↔ Forget password: Navigation by using `go` function, the location will be `/login` ↔ `/forget-password`
- Home ↔ Survey detail: Navigation by using `push` and `pop` function, the location will be `/` ↔ `/surveys/:surveyId`
- Survey detail ↔ Question detail: Navigation by using `go` function, the location will be `/surveys/:surveyId` ↔ `/questions/:surveyId?questionNumber=:questionNumber`
- Question detail ➡️ Home: Navigation by using `pop` function,  the location will be `/questions/:surveyId?questionNumber=:questionNumber` ➡️ `/` (Because if popping `questions` out, we get `/`)

Please take a look.

Reply:
It's fine for me 🙏 Please go ahead 🚀